src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -511,12 +511,13 @@ private static IEnumerable<ProjectItem> GetPackageReferences(Project project, Li
         private static IEnumerable<InstalledPackageReference> GetPackageReferencesFromTargets(string projectPath,
            string libraryName, string framework)
         {
+            var collectPackageReference = "CollectPackageReferences";
             var globalProperties = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
                 { { "TargetFramework", framework } };
             var newProject = new ProjectInstance(projectPath, globalProperties, null);
-            newProject.Build(new[] { "CollectPackageReferences" }, new List<Microsoft.Build.Framework.ILogger> { }, out var targetOutputs);
+            newProject.Build(new[] { collectPackageReference }, new List<Microsoft.Build.Framework.ILogger> { }, out var targetOutputs);
 
-            return targetOutputs.Single().Value.Items.Select(p =>
+            return targetOutputs.First(e => e.Key.Equals(collectPackageReference)).Value.Items.Select(p =>
```

Comment:
What's the reason that more than one exists? Using `First` rather than `Single` seems like it might be a hack to hide a different problem.

Reply:
InitialTargets/BeforeTargets/AfterTargets will all manifest in more than 1 target executing. 
I'm just using first because there's no way the enumeration will contain more than 1, and I don't need it to enumerate every result, just until it finds the right one.